app/components/submission/ManuscriptMetadata/SubjectAreaDropdown.test.js

```diff
@@ -0,0 +1,120 @@
+import React from 'react'
+import { shallow, mount } from 'enzyme'
+import theme from '@pubsweet/elife-theme'
+
+import SubjectAreaDropdown from './SubjectAreaDropdown'
+
+const onChange = jest.fn()
+
+const MyDropdown = (
+  <SubjectAreaDropdown
+    label="My label"
+    name="My name"
+    onBlur={jest.fn()}
+    onChange={onChange}
+    savedValues={[]}
+    theme={theme}
+  />
+)
+
+describe('SubjectAreaDropdown component', () => {
+  it('renders label & select', () => {
+    const wrapper = shallow(MyDropdown).dive() // dive is necessary because SubjectAreaDropdown is wrapped by withTheme
+    expect(wrapper.contains('My label')).toBe(true)
+    expect(wrapper.find('[inputId="subject-area-select"]').exists()).toEqual(
+      true,
+    )
+    expect(wrapper.contains('No more than two subject areas')).toBe(false)
+  })
+
+  describe('Integration with react-select', () => {
+    let wrapper, selectWrapper, selectInput
+
+    beforeEach(() => {
+      wrapper = mount(MyDropdown)
+      selectWrapper = wrapper.find('Select')
+      selectInput = selectWrapper.find('input')
+    })
+
+    const typeSubjectArea = letters => {
+      // We need 2 steps to programmatically simulate typing in a letter
+      // 1. Add a letter to the value of the input field (does not cause the dropdown to appear by itself)
+      selectWrapper.props().onInputChange(letters)
+      // 2. Open the dropdown
+      selectInput.simulate('change')
+    }
+
+    it('focusing causes the dropdown to open & display every option', () => {
+      typeSubjectArea('') // typing nothing into the field seems to be the only way to programmatically focus react-select)
+      expect(wrapper.find('Menu').exists()).toEqual(true)
+      expect(wrapper.find('Option')).toHaveLength(17)
+    })
+
+    it('an option can be selected from the dropdown', () => {
+      typeSubjectArea('b')
+      selectInput.simulate('keyDown', { keyCode: 9, key: 'Tab' })
+      expect(onChange.mock.calls).toHaveLength(1)
```

Comment:
You should probably clear this mock in a `beforeEach`. Otherwise this test could pass in error if a previous test calls `onChange` and this one doesn't.